app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,96 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :set_posts, only: [:index]
+  # GET /posts
+  # GET /posts.json
+  def index
+    #@posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show
+    @posts=Post.find(params[:id])
+    @posts=[@posts]
+  end
+
+  # GET /posts/new
+  def new
+    @post = Post.new
+  end
+
+  # GET /posts/1/edit
+  def edit
+  end
+
+  # POST /posts
+  # POST /posts.json
+
+  def create
+    @post = Post.new(post_params) do |post|
+      post.user = current_user
+      post.attachment=""
+    end
+    #debugger
+    if @post.save
+      redirect_to root_path
+    else
+      redirect_to root_path, notice: @post.errors.full_messages.first
+    end
```

Comment:
In case of error redirect to form.
In case of success redirect to post.

Reply:
Done